kornia/utils/draw.py

```diff
@@ -205,3 +207,106 @@ def draw_rectangle(
+        w: right most coordinate (left coordinate is assumed to be 0)
+    Returns:
+        The left and right edges of the polygon of shape (B,B).
+    """
+
+    dtype = polygon.dtype
+
+    # Check if polygons are in loop closed format, if not -> make it so
+    if not torch.allclose(polygon[..., -1, :], polygon[..., 0, :]):
+        polygon = torch.cat((polygon, polygon[..., :1, :]), dim=-2)  # (B, N+1, 2)
+
+    # Partition points into edges
+    x_start, y_start = polygon[..., :-1, 0], polygon[..., :-1, 1]
+    x_end, y_end = polygon[..., 1:, 0], polygon[..., 1:, 1]
+
+    # Create scanlines, edge dx/dy, and produce x values
+    ys = torch.arange(h, device=polygon.device, dtype=dtype)
+    dx = ((x_end - x_start) / (y_end - y_start + 1e-12)).clamp(-w, w)
+    xs = (ys[..., :, None] - y_start[..., None, :]) * dx[..., None, :] + x_start[..., None, :]
+
+    # Only count edge in their active regions (i.e between the vertices)
+    valid_edges = (y_start[..., None, :] <= ys[..., :, None]).logical_and(ys[..., :, None] <= y_end[..., None, :])
+    valid_edges |= (y_start[..., None, :] >= ys[..., :, None]).logical_and(ys[..., :, None] >= y_end[..., None, :])
+    x_left_edges = xs.clone()
+    x_left_edges[~valid_edges] = w
+    x_right_edges = xs.clone()
+    x_right_edges[~valid_edges] = -1
+
+    # Find smallest and largest x values for the valid edges
+    x_left = x_left_edges.min(dim=-1).values
+    x_right = x_right_edges.max(dim=-1).values
+    return x_left, x_right
+
+
+def _batch_polygons(polygons: List[Tensor]) -> Tensor:
+    r"""Converts a List of variable length polygons into a fixed size tensor.
+    Works by repeating the last element in the tensor.
+    Args:
+        polygon: List of variable length polygons of shape [N_1 x 2, N_2 x 2, ..., N_B x 2].
+                    B is the batch size,
+                    N_i is the number of points,
+                    2 is (x, y).
+    Returns:
+        A fixed size tensor of shape (B, N, 2) where N = max_i(N_i)
+    """
+    B, N = len(polygons), len(max(polygons, key=len))
+    batched_polygons = torch.zeros(B, N, 2, dtype=polygons[0].dtype, device=polygons[0].device)
+    for b, p in enumerate(polygons):
+        batched_polygons[b] = torch.cat((p, p[-1:].expand(N - len(p), 2))) if len(p) < N else p
+    return batched_polygons
+
+
+def draw_convex_polygons(images: Tensor, polygons: Union[Tensor, List[Tensor]], colors: Tensor) -> Tensor:
+    r"""Draws convex polygons on a batch of image tensors.
+
+    Args:
+        images: is tensor of BxCxHxW.
+        polygons: represents polygons as points, either BxNx2 or List of variable length polygons.
+            N is the number of points.
+            2 is (x, y).
+        color: a B x 3 tensor or 3 tensor with color to fill in.
+
+    Returns:
+        This operation modifies image inplace but also returns the drawn tensor for
+        convenience with same shape the of the input BxCxHxW.
+
+    Note:
```

Comment:
```suggestion
    .. note::
```
This will render better thanks to furo theme